python_on_whales/components/compose/cli_wrapper.py

```diff
@@ -252,7 +253,41 @@ def run(
         # volumes: bool = "todo",
         workdir: Union[None, str, Path] = None,
     ) -> Union[str, python_on_whales.components.container.cli_wrapper.Container]:
-        """Not yet implemented"""
+        """Run a one-off command on a service.
+
+        # Arguments
+            service: The name of the service.
+            command: The command to execute.
+            detach: if `True`, returns immediately with the Container.
+                    If `False`, returns the command stdout as string.
+            name: Assign a name to the container.
+            dependencies: Also start linked services.
+            publish: Publish a container's port(s) to the host.
+            service_ports: Enable service's ports and map them to the host.
+            remove: Automatically remove the container when it exits.
+            use_aliases: Use the service's network aliases in the connected network(s).
+            tty: Allocate a pseudo-TTY. Allow the process to access your terminal
+                to write on it.
+            stream: Similar to `docker.run(..., stream=True)`.
+            user: Username or UID, format: `"<name|uid>[:<group|gid>]"`
+            workdir: Working directory inside the container
+
+        # Returns:
+            Optional[str]
+
+        """
+
+        if tty and stream:
+            raise ValueError(
+                "You can't set tty=True and stream=True at the same"
+                "time. Their purpose are not compatible."
+            )
+
+        if detach and stream:
+            raise ValueError(
+                "You can't detach and stream at the same time. It's not compatible."
+            )
+
```

Comment:
I would also add here this:

```suggestion
            )

    if detach and tty:
            raise ValueError(
                "You can't detach and set tty=True at the same time. It's not compatible. "
                "Try setting tty=False in docker.compose.run(...)."
            )
```